### PR TITLE
IUO: Updates to PrintAsObjC to use the decl attribute for IUO.

### PR DESCRIPTION
**Pull request by rudkx**
Use the attribute to update the OptionalTypeKind used for various
decls that we print.

### CODE REVIEW

**Comment by rudkx** [comment]
@swift-ci Please smoke test

File: lib/PrintAsObjC/PrintAsObjC.cpp [hunk 1]
```diff
@@ -716,10 +739,17 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     if (params->size()) {
       interleave(*params,
                  [&](const ParamDecl *param) {
-                   print(param->getInterfaceType(), OTK_None, param->getName(),
+                   OptionalTypeKind optionalKind = OTK_None;
+                   Type objTy = param->getInterfaceType();
+                   if (FD->getAttrs()
+                           .hasAttribute<ImplicitlyUnwrappedOptionalAttr>()) {
```
Comment (jrose-apple) [inline lib/PrintAsObjC/PrintAsObjC.cpp]: This should be `param`, right?

**Review by jrose-apple — COMMENTED** [review]
Looks good, although I wonder if some of that can be factored out some. There's one place that I think is a typo, though. Can you add a test to catch this case?

File: lib/PrintAsObjC/PrintAsObjC.cpp (hunk 1 above)
Comment (rudkx) [reply lib/PrintAsObjC/PrintAsObjC.cpp]: Yeah that's clearly bogus, and also manages to not set optionalKind. I suspect we already have a test that would show this, but it requires the changes to actually remove the IUO type from the type system for the problem to show up.

I'll double check that before merging, though.

**Comment by rudkx** [comment]
This requires https://github.com/apple/swift/pull/13895 to be merged before tests will pass.

**Comment by rudkx** [comment]
@jrose-apple I think this is in pretty good shape now. I refactored the common bits and made sure we were hitting all modified code, adding new tests where they didn't already exist.

**Comment by rudkx** [comment]
@swift-ci Please smoke test

**Review by jrose-apple — APPROVED** [review]
Looks great. Thanks, Mark!

**Pull request merged**